lib/Default/SmrPlanet.class.inc

```diff
@@ -676,15 +676,12 @@ class SmrPlanet {
 			$this->db->query('
 				SELECT * 
 				FROM planet_is_building 
-				JOIN planet_can_build USING(construction_id)
 				WHERE game_id = ' . $this->db->escapeNumber($this->getGameID()) . ' 
-				AND sector_id = ' . $this->db->escapeNumber($this->getSectorID()). ' 
-				AND planet_type_id = '.$this->getTypeID());
+				AND sector_id = ' . $this->db->escapeNumber($this->getSectorID()));
 			
 			while($this->db->nextRecord()) {
 				if($this->db->getInt('time_complete') <= TIME) {
-					$PLANET_BUILDINGS =& Globals::getPlanetBuildings();
-					$expGain = $this->db->getInt('exp_gain');
+					$expGain = $this->db->getInt('construction_exp');
```

Comment:
That's what I coded first, but it had an error because `getConstructionExp` used `$this->db` to perform another query, so I would either need to:
* remove the db query in `getConstructionExp` (perhaps by the method you suggest below) or...
* grab all the variables from the previous query before calling `getConstructionExp`